Backported to 1.20.1

VipCoder8 opened the pull request:
- Fixed player inventory keybind issues if sprint set to toggle.
- Backported the excluding undead horses for swimming feature.
- Merged GirlInPurple's changes as they were not present in this branch.

VipCoder8 committed:
Backported to 1.20.1

- Fixed player inventory keybind issues if sprint set to toggle
- Backported the excluding undead horses for swimming feature
- Merged GirlInPurple's changes as they were not present in this branch.

VipCoder8 committed:
Backported to 1.20.1

- Fixed player inventory keybind issues if sprint set to toggle
- Backported the excluding undead horses for swimming feature
- Merged GirlInPurple's changes as they were not present in this branch.

CodeF53 commented on build.gradle:
```diff
@@ -38,7 +38,7 @@ dependencies {
-	//modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
+	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
 	modIncludeImplementation(fabricApi.module("fabric-api-base", project.fabric_version))
 	modIncludeImplementation(fabricApi.module("fabric-rendering-data-attachment-v1", project.fabric_version))
```
If you are depending on the entire fabric api, you should get rid of its submodules

CodeF53 commented on src/main/java/net/F53/HorseBuff/mixin/Client/InventoryAccessor.java:
```diff
@@ -21,13 +22,14 @@ public abstract class InventoryAccessor {
         assert this.player != null;
-        if (MinecraftClient.getInstance().options.sprintKey.isPressed()) {
+        boolean leftControl = GLFW.glfwGetKey(MinecraftClient.getInstance().getWindow().getHandle(), GLFW.GLFW_KEY_LEFT_CONTROL) == 1;
+        boolean rightControl = GLFW.glfwGetKey(MinecraftClient.getInstance().getWindow().getHandle(), GLFW.GLFW_KEY_RIGHT_CONTROL) == 1;
```
You aren't using the keybind you made for this in client init? 

Also hard coding to ctrl is horrid because 70% of mc playerbase rebinds crouch to ctrl (which dismounts the horse)

VipCoder8 replied:
I was told by Fyoncle(who asked me to fix the issue) that the player's inventory should be opened by holding control and pressing E, if a horse is mounted(default key combination), while users can rebind a key without the need of using specific combination.

CodeF53 commented on src/main/java/net/F53/HorseBuff/utils/KeyInput.java:
This entire file isn't necessary, just use the exact keybind code found in main or girlinpurples pr

VipCoder8 replied:
Based on my first reply it is necessary.

CodeF53 reviewed (changes requested):
Please also update and run the github workflow so I can test the the compiled binary

VipCoder8 committed:
Backported to 1.20.1

- Fixed player inventory keybind issues if sprint set to toggle
- Backported the excluding undead horses for swimming feature
- Merged GirlInPurple's changes as they were not present in this branch.

VipCoder8 committed:
Update HorseBuffInit.java

Remove accidentally added unused import.

VipCoder8 committed:
Updated workflow.

VipCoder8 committed:
Merge remote-tracking branch 'origin/1.20.1' into 1.20.1

CodeF53 reviewed (approved):
Still needs an updated workflow file but otherwise lgtm

CodeF53 commented:
wait the flow was updated, then why did the build fail
